### ISSUE TITLE
Error with module keras/metrics/accuracy_metrics.py and tensorflow module tf.keras.metrics.Accuracy()

### ISSUE THREAD

**Issue by Mr-Nobody-dey** [keras-team/keras#18796]
In my model metrics parameter when I use "tf.keras.metrics.Accuracy()", I get an output of 0.0 for every epoch. But changing it to 'accuracy' outputs some floating point number, which looks correct.

Is their anything wrong with tf.keras.metrics.Accuracy() or may be with my code?

**Comment by SuryanarayanaY** [comment]
Hello @Mr-Nobody-dey ,

I tried a sample model in **keras-nightly(3.0.0.Dev)** with `metrics=['accuracy']` and `metrics=[keras.metrics.Accuracy()] ` separately and observed that when `metrics = 'accuracy'` got accuracy as `0.49` but with `keras.metrics.Accuracy()` getting accuracy as `0.0000e+00` . Attached [gist](https://colab.sandbox.google.com/gist/SuryanarayanaY/fd0d013978724b266163013c54129299/18796_sample_model-keras-nightly-3-0-dev.ipynb) for reference. Could you please confirm whether this is what you are trying to report?

This needs to be looked into for this behaviour with `metrics=[keras.metrics.Accuracy()] `

**Comment by fchollet** [comment]
> In my model metrics parameter when I use "tf.keras.metrics.Accuracy()", I get an output of 0.0 for every epoch. But changing it to 'accuracy' outputs some floating point number, which looks correct.

`"accuracy"` gets converted to either `BinaryAccuracy` or `CategoricalAccuracy` based on your output shape. Just specify one of these 2 instead of `Accuracy` (which used direct value comparison). You probably need `BinaryAccuracy`.
